### PR TITLE
docs: update spl token exchange integration

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

SPL Token exchange integration docs are outdated.  Missing best practices use of ATAs, token balance checks and modern instruction names

#### Summary of Changes

Update them

### CODE REVIEW

**Review by aeyakovenko — DISMISSED** [review]

**Comment by aeyakovenko** [comment]
this goes into way more details.  are we at the point that we can do the same for tokens?

https://docs.solana.com/integrations/exchange#poll-for-blocks

File: docs/src/integrations/exchange.md [hunk 1]
```diff
@@ -694,35 +694,38 @@ Signature: 3R6tsog17QM8KfzbcbdP4aoMfwgo6hBggJDVy7dZPVmH2xbCWjEj31JKD53NzMrf25ChF
...
 the exchange should check the address as
 [described above](#validating-user-supplied-account-addresses-for-withdrawals).
-
-From the withdrawal address, the associated token account for the correct mint
-determined and the transfer issued to that account. Note that it's possible
-that the associated token account does not yet exist, at which point the
+Additionally it is recommended that the account at this address be owned by the
+System Program and that its SOL balance be non-zero.  Non-system owned withdrawal
+addresses should be rejected and zero-SOL-balance accounts should require user
+confirmation before proceeding.
```
Comment (mvines) [inline docs/src/integrations/exchange.md:723]:
```suggestion
Additionally this address must be owned by the System Program and have no account data.  If the address has no SOL balance, user confirmation should be obtained before proceeding with the withdrawal.  All other withdrawal addresses must be rejected.
```

**Review by mvines — APPROVED** [review]
I made some minor changes to strengthen the language so that nobody can come back in the future with "you said it was optional and something bad happened to me"

**Comment by t-nelson** [comment]
> this goes into way more details. are we at the point that we can do the same for tokens?
> 
> https://docs.solana.com/integrations/exchange#poll-for-blocks

Yeah we [reference](https://github.com/solana-labs/solana/blob/master/docs/src/integrations/exchange.md?plain=1#L702) this section from the SPL Token deposits section

**Comment by t-nelson** [comment]
> I made some minor changes to strengthen the language so that nobody can come back in the future with "you said it was optional and something bad happened to me"

Good call!  We'll inevitably get some push back, but we can handle that on a case by case basis

**Pull request merged**